python/ray/data/datasource/file_based_datasource.py

```diff
@@ -237,6 +244,32 @@ def expand_paths(
         return expanded_paths, file_sizes
 
 
+class FastFileMetadataProvider(DefaultFileMetadataProvider):
+    """Fast Metadata provider for FileBasedDatasource implementations.
+
+    Offers improved performance vs. DefaultFileMetadataProvider by skipping directory
+    path expansion and file size collection. While this performance improvement may be
+    negligible for local filesystems, it can be substantial for cloud storage service
+    providers.
+
+    This should only be used when all input paths are known to be files.
+    """
+
+    def expand_paths(
+        self,
+        paths: List[str],
```

Comment:
Do all paths need to be for the same block?
If not, why does it make sense to return BlockMetadata as value?

Reply:
I assume this comment is meant to apply to the core `__call__` and `_get_block_metadata` methods, which do require that all paths provided are part of the same block. I think this could be made more clear with some doc string updates.

Reply:
I've added some clarification to the docstrings here in the latest commit.

Reply:
Thanks.
@clarkzinzow Is it true that we have multiple files to load into a single block?